src/Nethermind/Nethermind.State.Test.Runner/StateTestTxTracer.cs

```diff
@@ -91,6 +91,32 @@ public void ReportOperationRemainingGas(long gas)
         }
     }
 
+    public void SetOperationStack(TraceStack stack)
+    {
+        _traceEntry.Stack = new List<string>();
+        foreach (string s in stack.ToHexWordList())
+        {
+            ReadOnlySpan<char> inProgress = s.AsSpan();
+            if (s.StartsWith("0x"))
+            {
+                inProgress = inProgress.Slice(2);
+            }
+
+            inProgress = inProgress.TrimStart('0');
+
+            _traceEntry.Stack.Add(inProgress.Length == 0 ? "0x0" : "0x" + inProgress.ToString());
+        }
+
+    }
+
+    public void SetOperationMemory(TraceMemory memoryTrace)
+    {
+        if (IsTracingDetailedMemory)
+        {
+            _traceEntry.Memory = string.Concat("0x", string.Join("", memoryTrace.ToHexWordList().Select(mt => mt.Replace("0x", string.Empty))));
```

Comment:
Might be nicer to move this to a `ToHexString()` method for `TraceMemory`

Reply:
This sounds like it is the only place it is needed and it is not even in main solution but in test solution. Let's leave it here until needed in more places.